src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -169,22 +170,31 @@ trait SamRecord {
 
   @inline final def mateStart: Int = getMateAlignmentStart
   @inline final def mateStart_=(s: Int):Unit = setMateAlignmentStart(s)
+
+  @inline final def mateCigar: Option[Cigar] = {
+    require(paired && mateMapped, "Cannot get a mate cigar on read without a mapped mate.")
+    get[String]("MC").map(Cigar.apply)
+  }
   @inline final def mateEnd: Option[Int] = {
     require(paired && mateMapped, "Cannot get mate end position on read without a mapped mate.")
-    get[String]("MC").map(cig => mateStart + Cigar(cig).lengthOnTarget - 1)
+    mateCigar.map(_.lengthOnTarget + mateStart - 1)
   }
-
   @inline final def mateUnclippedStart: Option[Int] = {
     require(paired && mateMapped, "Cannot get mate unclipped start position on read without a mapped mate.")
-    get[String]("MC").map(cig => mateStart - Cigar(cig).iterator.takeWhile(_.operator.isClipping).map(_.length).sum)
+    mateCigar.map(cigar => mateStart - cigar.iterator.takeWhile(_.operator.isClipping).map(_.length).sum)
   }
   @inline final def mateUnclippedEnd: Option[Int] = {
     require(paired && mateMapped, "Cannot get mate unclipped end position on read without a mapped mate.")
-    get[String]("MC").map { cig =>
-      val cigar = Cigar(cig)
+    mateCigar.map { cigar =>
       mateStart + cigar.lengthOnTarget - 1 + cigar.reverseIterator.takeWhile(_.operator.isClipping).map(_.length).sum
     }
   }
+  @inline final def matesOverlap: Option[Boolean] = {
```

Comment:
1. mateOverlaps? I don’t see how this works on multiple mates :0
2. I prefer exception if the read pair doesn’t have a mate to be consistent with the current implementation. Happy to have another PR address the exception vs pure function issue to being the discussion there. If you and @tfenne object, let’s take this offline.

Reply:
Both these were caused by suggestions of Tim: https://github.com/fulcrumgenomics/fgbio/pull/717#discussion_r716027927

I think "mates overlap" was Tim's choice because the read pair's are each others mates.

As in, "do mates overlap? Yes, mates overlap."

Let me know what you'd like me to do!

Reply:
I disagree with @tfenne so let's take this offline.  Since I don't like the new name ( E.g. "does I overlap my mate", "yes, my mate overlaps") and I think we should keep the implementation consistent.